Add mental model section and FAQ on S3 object versioning (#1485)
quiltdata/quilt: docs/FAQ.md

> ## How does Quilt play with S3 object versioning?

Author: ```suggestion
## How does Quilt versioning relate to S3 object versioning?
```

Quilt packages are one level of abstraction above S3 object versions.
Object versions track mutations to a single file,

> whereas a quilt package may reference one or more files and makes this entire collection of files reproducible.

Author: ```suggestion
whereas a quilt package references an collection files and assigns this collection a unique version.
```
